components/service/contile/src/main/java/mozilla/components/service/contile/ContileTopSitesProvider.kt

```diff
@@ -0,0 +1,84 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.contile
+
+import mozilla.components.concept.fetch.Client
+import mozilla.components.concept.fetch.Request
+import mozilla.components.concept.fetch.isSuccess
+import mozilla.components.feature.top.sites.TopSite
+import mozilla.components.feature.top.sites.TopSitesProvider
+import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.ktx.android.org.json.asSequence
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.IOException
+
+internal const val CONTILE_ENDPOINT_URL = "https://contile.services.mozilla.com/v1/tiles"
+
+/**
+ * Provide access to the Contile services API.
+ *
+ * @property client [Client] used for interacting with the Contile HTTP API.
+ */
+class ContileTopSitesProvider(
+    private val client: Client
+) : TopSitesProvider {
+
+    private val logger = Logger("ContileTopSitesProvider")
+
+    @Throws(IOException::class)
+    override suspend fun getTopSites(): List<TopSite.Provided> {
+        return try {
+            fetchTopSites()
+        } catch (e: IOException) {
+            logger.error("Failed to fetch contile top sites", e)
+            throw e
+        }
+    }
+
+    private fun fetchTopSites(): List<TopSite.Provided> {
+        client.fetch(
+            Request(url = CONTILE_ENDPOINT_URL)
```

Comment:
This endpoint could be good as a parameter with `CONTILE_ENDPOINT_URL` as the default so we aren't tied to this endpoint (say, if we need a separate one for Focus/Fenix).